vpr/src/place/median_move_generator.cpp

```diff
@@ -240,77 +266,84 @@ static void get_bb_from_scratch_excluding_block(ClusterNetId net_id, t_bb* bb_co
  * the pins always lie on the outside of the bounding box.            *
  * The x and y coordinates are the pin's x and y coordinates.         */
 /* IO blocks are considered to be one cell in for simplicity.         */
-static bool get_bb_incrementally(ClusterNetId net_id, t_bb* bb_coord_new, int xold, int yold, int xnew, int ynew) {
+static bool get_bb_incrementally(ClusterNetId net_id, t_bb& bb_coord_new, int xold, int yold, int xnew, int ynew, int /* layer */) {
```

Comment:
Watch that there are no loops that go over every terminal on a high fanout net and can be called once per move.

Reply:
Delete layer and comment that this is updating only the x and y parts of the bounding box.

Reply:
I have rechecked and ensured that I am not iterating over all terminals. In this move, I have implemented the use of a 3D bounding box, regardless of the type chosen by the user, to simplify the process (The bounding box type specified by the user is utilized by the WL estimator.)
Since this implementation relies on the bounding box stored in the placement context, I need to create a 3D bounding box if the type stored there is "per-layer." This is why the function "union_2d_bb_incr" is called when the bounding box type is "per-layer." In that function, I iterate over all layers, not terminals, to create a 3D bounding box from the "per-layer" bounding box.